File: skillful_network_server/src/main/java/fr/uca/cdr/skillful_network/entities/application/TrainingApplication.java
package fr.uca.cdr.skillful_network.entities.application;

import com.fasterxml.jackson.annotation.JsonIgnore;
import fr.uca.cdr.skillful_network.entities.user.User;

import javax.persistence.*;
import java.util.Date;

@Entity
public class TrainingApplication extends Application {

    @ManyToOne(cascade = CascadeType.ALL)
//    @JoinColumn(name = "training_id")
    @JsonIgnore
    private Training offer;

    public TrainingApplication() {

    }

    public TrainingApplication(User user, Training offer) {
        this.offer = offer;
        this.user = user;
    }

    public TrainingApplication(User user, ApplicationStatus status, Date submitDate, Training offer) {
        super(user, status, submitDate);
        this.offer = offer;
    }

    public Training getOffer() {
        return offer;
    }

    public void setOffer(Training offer) {
        this.offer = offer;
    }
}

File: skillful_network_client/src/app/shared/models/application/training-application.ts
import { Application } from './application';
import { Training } from './training';


export class TrainingApplication extends Application {

    private _training: Training;

    constructor(data: any) {
       super(data);
       this.training = data.offer;
    }

    public set training(training: Training) {
        this.training = training;
    }

    public get training() : Training {
        return this.training;
    }

}

